tests/queries/tests.py

```diff
@@ -3950,3 +3951,29 @@ def test_ticket_23622(self):
             set(Ticket23605A.objects.filter(qy).values_list('pk', flat=True))
         )
         self.assertSequenceEqual(Ticket23605A.objects.filter(qx), [a2])
+
+
+class Ticket20393Tests(TestCase):
```

Comment:
I think "ticket X tests" is a pattern we want to phase out - it's not a very useful name

How about `ReprTests` ?

Also your test method is really many different tests, separated by comments - can you split them into individual methods like `test_unevaluated`, `test_evaluated`, `test_stale_data`? Then you can drop the comments. Also if you're feeling good you can move the creation of the set of `Individual` objects to `setUpTestData`, and use `bulk_create`, to make it faster and shared.